db-service/lib/cqn2sql.js

```diff
@@ -492,12 +494,25 @@ class CQN2SQLRenderer {
       return (this.sql = `INSERT INTO ${this.quote(entity)}${alias ? ' as ' + this.quote(alias) : ''} (${this.columns.map(c => this.quote(c))}) VALUES (${columns.map(param)})`)
     }
 
-    const extractions = this.managed(
-      columns.map(c => ({ name: c })),
-      elements,
-      !!q.UPSERT,
-    )
-    const extraction = extractions.map(c => c.sql)
+    const extractions = this.managed(columns.map(c => ({ name: c })), elements)
+    const extraction = extractions
+      .map(c => {
+        const element = elements?.[c.name]
+        if (element?.['@cds.extension']) {
+          return false
+        }
+        if (c.name === 'extensions__') {
+          const merges = extractions.filter(c => elements?.[c.name]?.['@cds.extension'])
+          if (merges.length) {
+            c.sql = `json_set(ifnull(${c.sql},'{}'),${merges.map(
+              c => this.string('$."' + c.name + '"') + ',' + c.sql,
+            )})`
+          }
+        }
+        return c
+      })
+      .filter(a => a)
```

Comment:
All this is for the UIFlex thing which we removed in cds8, isn't it? → why do we still need that?

Reply:
This code was put in here to make `mtxs` pass its tests. My understanding is that in the mean time someone looked into the root cause of this arriving at the database service. Therefor it should be possible to remove this logic from the database now.

After applying all review comments the internal tests will be run with this PR to verify this.